catalyst-gateway/bin/src/service/common/auth/rbac/scheme.rs

```diff
@@ -153,23 +151,18 @@ async fn checker_api_catalyst_auth(
         return Ok(token);
     };
 
-    // Step 6: get and build latest registration chain from the db.
-    let reg_chain = match token.reg_chain().await {
-        Ok(Some(reg_chain)) => reg_chain,
-        Ok(None) => {
-            debug!(
-                "Unable to find registrations for {} Catalyst ID",
-                token.catalyst_id()
-            );
-            return Err(AuthTokenError::RegistrationNotFound.into());
-        },
-        Err(err) if err.is::<CassandraSessionError>() => {
-            return Err(ServiceUnavailableError(err).into())
-        },
-        Err(err) => {
-            debug!("Unable to build a registration chain Catalyst ID: {err:?}");
-            return Err(AuthTokenError::BuildRegChain(err.to_string()).into());
-        },
+    // Step 6: get the registration chain
+    // Get the registration from the persistent first, if not found, try volatile.
+    let reg_chain = RBAC_CACHE
+        .get(token.catalyst_id(), true)
+        .or_else(|| RBAC_CACHE.get(token.catalyst_id(), false));
```

Comment:
I'm not sure about that. As far as I understand, it is allowed to use volatile registrations everywhere. In that case I suppose we should be looking at volatile registrations first.

@stevenj @Mr-Leshiy

Reply:
Yes you are correct, we should always look at the volatile first.